### PR TITLE
Implement financial list removal

### PR DESCRIPTION
**Pull request by tejohnso**
@settinghead please review
@Rise-Vision/delivery FYI

### CODE REVIEW

File: src/components/financialLists/financialLists.controller.js [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
+  constructor( financialListAddService, financialListRemoveService, $state, authService, displayUsersService, $async ) {
     "ngInject";
 
     authService.redirectIfNotLoggedIn();
 
+    this.financialLists = [
+      //{id: "-KZSdjLT_gc-ezDbT1j2", name: "testname", lastModified: Date.now(), modifiedBy: "Someone"}
+    ];
+
+    this.$onInit = $async( async() => {
```
Comment (settinghead) [inline src/components/financialLists/financialLists.controller.js]: I don't think you need the `$async` wrapper and the `async` & `await` keywords here, unless you want to use a series of `await`'s for sequential operations.

File: src/components/financialLists/financial_list_remove.service.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    return {
+      remove( listId, displayId ) {
+        let deferred = $q.defer(),
+          update = {};
+
+        update[ "/lists/" + listId ] = null;
+        update[ "/displays/" + displayId + "/lists/" + listId ] = null;
+
+        root.update( update )
```
Comment (settinghead) [inline src/components/financialLists/financial_list_remove.service.js]: It looks like `return root.update(...)` directly (without creating the additional promise) would achieve the same result.

**Comment by tejohnso** [comment]
@settinghead thanks - I've cleaned it up in #2597457

**Comment by settinghead** [comment]
@tejohnso looks good now :+1:

**Pull request merged**